src/content/api/plugins.md

```diff
@@ -99,6 +99,35 @@ compiler.hooks.myCustomHook.call(a, b, c);
 Again, see the [documentation](https://github.com/webpack/tapable) for `tapable` to learn more about the
 different hook classes and how they work.
 
+## Reporting Progress
+
+Plugins can report progress information, which is printed by [the CLI](/api/cli/) when the `--progress` argument is used. More generally, progress information is passed to [`ProgressPlugin`](/plugins/progress-plugin/), which can be used to customize how progress information is reported to the user.
+
+To report progress, a plugin must `tap` into a hook using the `context: true` option:
+
+```js
+compiler.hooks.emit.tapAsync({
+  name: 'MyPlugin',
+  context: true
+}, (context, compiler, callback) => {
+  const reportProgress = context && context.reportProgress;
+  if (reportProgress) reportProgress(0.95, 'Starting work');
+  setTimeout(() => {
+    if (reportProgress) reportProgress(0.95, 'Done work');
+    callback();
+  }, 1000);
+});
+```
+
+The `reportProgress` function has this signature:
+
+```js
+reportProgress(percentage: number, ...args: string[]): void;
```

Comment:
Please use JS.

Reply:
Alternatively, add basic JSDoc notation or a link to its Source code.

Reply:
Now updated to use JS